packages/builder/src/templates/generic/useHttpMutation.hbs

```diff
@@ -51,9 +51,17 @@ const useHttpMutation = <Req, Res>(params: UseMutationParams<Res>) => {
       if (response.ok) {
         return response.json() as Promise<Res>;
       }
+      if (response.headers.get('content-type')?.includes('text/html')) {
+        const result = await response.text();
+        throw new QueryError(
+          getQueryErrorCode(response.status),
+          result
+        );
+      }
+      const result = await response.json();
       throw new QueryError(
         getQueryErrorCode(response.status),
-        response.statusText
+        result?.message || result?.error || response.statusText
```

Comment:
please send the `result` instead because you know is the JSON, you are doing  `.json()` and it's up to the client and server to handle the structure the JSON is returning 

```
 throw new QueryError(
        getQueryErrorCode(response.status),
        result,
```